### PR TITLE
added azure-devops-agent chart

### PR DESCRIPTION
**Pull request by chrisorisawayi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-231

### Change description ###
Added chart for azure DevOps agent

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: stable/azure-devops-agent/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{{ if .Values.secrets.create }}
```
Comment (timja) [inline stable/azure-devops-agent/templates/secret.yaml:1]: shouldn't this be:
```suggestion
{{ if .Values.azpToken }}
```

File: stable/azure-devops-agent/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (timja) [inline stable/azure-devops-agent/templates/service.yaml:1]: does this need a service?

Comment (chrisorisawayi) [reply stable/azure-devops-agent/templates/service.yaml:1]: Service not needed.  Agent communication is outbound poll only to Azure pipelines.  Will remove.

File: stable/azure-devops-agent/templates/secret.yaml (hunk 1 above)
Comment (chrisorisawayi) [reply stable/azure-devops-agent/templates/secret.yaml:1]: Looking this again, I can't even see the purpose for the secret if it's expected to pull the PAT from the values file (can't see how the PAT will be provided from the value in a secure way).  

Having a rethink on this and will test/add updates to read PAT from keyvault using CSI.  Will cancel and issue a new PR once tested.

Comment (timja) [reply stable/azure-devops-agent/templates/secret.yaml:1]: sure

**Pull request closed without merging**